src/model/APNSRequest.php

```diff
@@ -48,32 +48,32 @@ function getUrlForConfiguration( APNSConfiguration $configuration ): string {
 	}
 
 	/**
-	 * @return (int|mixed|null|string)[]
+	 * @return string[]
 	 *
-	 * @psalm-return array{authorization: string, content-type: string, content-length: int, apns-expiration: int, apns-push-type: string, apns-topic: mixed, user-agent?: null|string, apns-priority?: mixed, apns-id?: string, apns-collapse-id?: null|string}
+	 * @psalm-return array<string, string>
```

Comment:
Reading the docs for the this syntax, which I know now is a [generic array](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#generic-arrays), I also saw Psalm allows defining non-empty generic arrays, with `non-empty-array<TKey, TValue>`. I'm not sure whether we could use this, but it's super cool.

Reply:
That's interesting! It definitely should be a `non-empty-array`, so could be worth adding for sure!

Nice catch!